pkg/k8s/validate.go

```diff
@@ -0,0 +1,278 @@
+package k8s
+
+import (
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation"
+
+	"github.com/celestiaorg/knuu/pkg/errors"
+)
+
+func validateDNS1123Label(name string, err *errors.Error) error {
+	if errs := validation.IsDNS1123Label(name); len(errs) > 0 {
+		return err.WithParams(name, errs)
+	}
+	return nil
+}
+
+func validateDNS1123Subdomain(name string, returnErr *errors.Error) error {
+	if errs := validation.IsDNS1123Subdomain(name); len(errs) > 0 {
```

Comment:
Does this also check the length of the label? In Kubernetes, there is a max limit.

Reply:
yes it does